Source/WebCore/replay/UserInputBridge.cpp

```diff
@@ -41,123 +41,129 @@
 
 namespace WebCore {
 
-UserInputBridge::UserInputBridge(Page& page)
-    : m_page(page)
+UserInputBridge::UserInputBridge(LocalFrame& frame)
+    : m_frame(frame)
 {
 }
 
 #if ENABLE(CONTEXT_MENU_EVENT)
-bool UserInputBridge::handleContextMenuEvent(const PlatformMouseEvent& mouseEvent, LocalFrame& frame, InputSource)
+bool UserInputBridge::handleContextMenuEvent(const PlatformMouseEvent& mouseEvent, InputSource)
 {
-    return frame.eventHandler().sendContextMenuEvent(mouseEvent);
+    return m_frame.eventHandler().sendContextMenuEvent(mouseEvent);
 }
 #endif
 
 bool UserInputBridge::handleMousePressEvent(const PlatformMouseEvent& mouseEvent, InputSource)
 {
-    auto* localMainFrame = dynamicDowncast<LocalFrame>(m_page.mainFrame());
-    if (!localMainFrame)
-        return false;
-    return Ref(*localMainFrame)->eventHandler().handleMousePressEvent(mouseEvent);
+    return m_frame.eventHandler().handleMousePressEvent(mouseEvent);
 }
 
 bool UserInputBridge::handleMouseReleaseEvent(const PlatformMouseEvent& mouseEvent, InputSource)
 {
-    auto* localMainFrame = dynamicDowncast<LocalFrame>(m_page.mainFrame());
-    if (!localMainFrame)
-        return false;
-    return Ref(*localMainFrame)->eventHandler().handleMouseReleaseEvent(mouseEvent);
+    return m_frame.eventHandler().handleMouseReleaseEvent(mouseEvent);
 }
 
 bool UserInputBridge::handleMouseMoveEvent(const PlatformMouseEvent& mouseEvent, InputSource)
 {
-    auto* localMainFrame = dynamicDowncast<LocalFrame>(m_page.mainFrame());
-    if (!localMainFrame)
-        return false;
-    return Ref(*localMainFrame)->eventHandler().mouseMoved(mouseEvent);
+    return m_frame.eventHandler().mouseMoved(mouseEvent);
 }
 
 bool UserInputBridge::handleMouseMoveOnScrollbarEvent(const PlatformMouseEvent& mouseEvent, InputSource)
 {
-    auto* localMainFrame = dynamicDowncast<LocalFrame>(m_page.mainFrame());
-    if (!localMainFrame)
-        return false;
-    return Ref(*localMainFrame)->eventHandler().passMouseMovedEventToScrollbars(mouseEvent);
+    return m_frame.eventHandler().passMouseMovedEventToScrollbars(mouseEvent);
 }
 
 bool UserInputBridge::handleMouseForceEvent(const PlatformMouseEvent& mouseEvent, InputSource)
 {
-    auto* localMainFrame = dynamicDowncast<LocalFrame>(m_page.mainFrame());
-    if (!localMainFrame)
-        return false;
-    return Ref(*localMainFrame)->eventHandler().handleMouseForceEvent(mouseEvent);
+    return m_frame.eventHandler().handleMouseForceEvent(mouseEvent);
 }
 
 bool UserInputBridge::handleKeyEvent(const PlatformKeyboardEvent& keyEvent, InputSource)
 {
-    return Ref(m_page.focusController().focusedOrMainFrame())->eventHandler().keyEvent(keyEvent);
+    auto* page = m_frame.page();
```

Comment:
It might make sense to not take in the frame here since we do `focusedOrMainFrame` and might even do that with site isolated key events.

Or maybe we can calculate focusedOrMainFrame in the UI process before sending the message to have the right frame here.

Reply:
It would seem better if we could make sure this is always being called on the correct frame instead of checking in this function, I can look at how I could do that.

We will still need to make sure `focusedOrMainFrame()` works correctly for site isolation, I didn't try to fix that in this patch.